diff --git a/plugins/save-param/xrandr-output.cpp b/plugins/save-param/xrandr-output.cpp
index 798d5a2..427aaa0 100644
--- a/plugins/save-param/xrandr-output.cpp
+++ b/plugins/save-param/xrandr-output.cpp
@@ -30,6 +30,7 @@
 #include <QRect>
 #include <QStandardPaths>
 #include <QtXml>
+#include <KF5/KScreen/kscreen/mode.h>
 #include <KF5/KScreen/kscreen/output.h>
 #include <KF5/KScreen/kscreen/edid.h>
 #include "xrandr-config.h"
diff --git a/plugins/xrandr/xrandr-manager.cpp b/plugins/xrandr/xrandr-manager.cpp
index e30ffe3..9f44af3 100644
--- a/plugins/xrandr/xrandr-manager.cpp
+++ b/plugins/xrandr/xrandr-manager.cpp
@@ -687,10 +687,10 @@ void XrandrManager::doOutputRemoved(int outputId)
     m_isInitFinish = false;
 }
 
-void XrandrManager::doPrimaryOutputChanged(const KScreen::OutputPtr &output)
+void XrandrManager::doPrimaryOutputChanged()
 {
-    if (output != nullptr)//if change output is disable, the output are nullptr
-        USD_LOG_SHOW_OUTPUT(output);
+//    if (output != nullptr)//if change output is disable, the output are nullptr
+//        USD_LOG_SHOW_OUTPUT(output);
 //    USD_LOG(LOG_DEBUG,"%s--------->%d",output->name().toLatin1().data(), output->isPrimary());
 }
 
@@ -1015,7 +1015,7 @@ int XrandrManager::initAllOutputs()
             }
         },Qt::QueuedConnection);
 
-        connect(output.data(), &KScreen::Output::isPrimaryChanged, this, [this](){
+        connect(output.data(), &KScreen::Output::priorityChanged, this, [this](){
             KScreen::Output *senderOutput = static_cast<KScreen::Output*> (sender());
             if (senderOutput == nullptr || UsdBaseClass::isWaylandWithKscreen() || !senderOutput->isEnabled()) {
                 USD_LOG(LOG_DEBUG,"had a bug..");
@@ -1199,7 +1199,7 @@ int XrandrManager::initAllOutputs()
     connect(m_configPtr.data(), SIGNAL(outputRemoved(int)),
             this, SLOT(doOutputRemoved(int)));
 
-    connect(m_configPtr.data(), &KScreen::Config::primaryOutputChanged,
+    connect(m_configPtr.data(), &KScreen::Config::prioritiesChanged,
             this, &XrandrManager::doPrimaryOutputChanged);
 
     if (m_xrandrSettings->keys().contains("hadmate2kscreen")) {//兼容mate配置
diff --git a/plugins/xrandr/xrandr-manager.h b/plugins/xrandr/xrandr-manager.h
index d2cf254..36040da 100644
--- a/plugins/xrandr/xrandr-manager.h
+++ b/plugins/xrandr/xrandr-manager.h
@@ -34,6 +34,7 @@
 
 
 #include <KF5/KScreen/kscreen/config.h>
+#include <KF5/KScreen/kscreen/mode.h>
 #include <KF5/KScreen/kscreen/log.h>
 #include <KF5/KScreen/kscreen/output.h>
 #include <KF5/KScreen/kscreen/edid.h>
@@ -94,7 +95,7 @@ public Q_SLOTS:
     void doRotationChanged(const QString &rotation);
     void doOutputAdded(const KScreen::OutputPtr &output);
     void doOutputRemoved(int outputId);
-    void doPrimaryOutputChanged(const KScreen::OutputPtr &output);
+    void doPrimaryOutputChanged();
 
     void setOutputsMode(QString modeName);
     void setOutputsParam(QString screensParam);
diff --git a/plugins/xrandr/xrandr-output.cpp b/plugins/xrandr/xrandr-output.cpp
index d95e7e8..86572d4 100644
--- a/plugins/xrandr/xrandr-output.cpp
+++ b/plugins/xrandr/xrandr-output.cpp
@@ -30,6 +30,7 @@
 #include <QRect>
 #include <QStandardPaths>
 #include <QtXml>
+#include <KF5/KScreen/kscreen/mode.h>
 #include <KF5/KScreen/kscreen/output.h>
 #include <KF5/KScreen/kscreen/edid.h>
 #include "xrandr-config.h"
