diff --git a/plugins/system/display/outputconfig.h b/plugins/system/display/outputconfig.h
index 89d61f0b..afe86424 100644
--- a/plugins/system/display/outputconfig.h
+++ b/plugins/system/display/outputconfig.h
@@ -5,6 +5,7 @@
 #include <QComboBox>
 #include <QWidget>
 
+#include <KF5/KScreen/kscreen/mode.h>
 #include <KF5/KScreen/kscreen/output.h>
 
 #include <QGSettings>
diff --git a/plugins/system/display/resolutionslider.cpp b/plugins/system/display/resolutionslider.cpp
index e0bd3d36..3a9b88fa 100644
--- a/plugins/system/display/resolutionslider.cpp
+++ b/plugins/system/display/resolutionslider.cpp
@@ -10,6 +10,7 @@
 
 #include <QDBusInterface>
 
+#include <KF5/KScreen/kscreen/mode.h>
 #include <KF5/KScreen/kscreen/output.h>
 
 static bool sizeLessThan(const QSize &sizeA, const QSize &sizeB)
diff --git a/plugins/system/display/widget.h b/plugins/system/display/widget.h
index bb09b15..4efbe3a 100644
--- a/plugins/system/display/widget.h
+++ b/plugins/system/display/widget.h
@@ -144,8 +144,8 @@ private Q_SLOTS:
     void outputAdded(const KScreen::OutputPtr &output, bool connectChanged);
     void outputRemoved(int outputId, bool connectChanged);
     void primaryOutputSelected(int index);
-    void primaryOutputChanged(const KScreen::OutputPtr &output);
+    void primaryOutputChanged();

     void showNightWidget(bool judge);

     void primaryButtonEnable(bool);             // 按钮选择主屏确认按钮
diff --git a/plugins/system/display/widget.cpp b/plugins/system/display/widget.cpp
index 40b4aa3..06d830c 100644
--- a/plugins/system/display/widget.cpp
+++ b/plugins/system/display/widget.cpp
@@ -196,7 +196,7 @@ void Widget::setConfig(const KScreen::ConfigPtr &config, bool showBrightnessFram
         outputRemoved(outputId, false);
     });
 
-    connect(mConfig.data(), &KScreen::Config::primaryOutputChanged,
+    connect(mConfig.data(), &KScreen::Config::prioritiesChanged,
             this, &Widget::primaryOutputChanged);
 
     for (const KScreen::OutputPtr &output : mConfig->outputs()) {
@@ -1282,9 +1282,10 @@ void Widget::primaryOutputSelected(int index)
 }

 // 主输出
-void Widget::primaryOutputChanged(const KScreen::OutputPtr &output)
+void Widget::primaryOutputChanged()
 {
     Q_ASSERT(mConfig);
+    const KScreen::OutputPtr &output = mConfig->primaryOutput();
     int index = output.isNull() ? 0 : monitorComboBox->findData(output->id());
