diff --git a/registeredQDbus/sysdbusregister.cpp b/registeredQDbus/sysdbusregister.cpp
index a69917d..d64efac 100644
--- a/registeredQDbus/sysdbusregister.cpp
+++ b/registeredQDbus/sysdbusregister.cpp
@@ -857,7 +857,7 @@ void SysdbusRegister::_getDisplayInfoThread()
                         display.edidHash = edidHash;
                         display.I2C_busType = QString::number(dlist_loc->info[i].path.path.i2c_busno);
                         ddca_create_edid_display_identifier(dlist_loc->info[i].edid_bytes,&did);
-                        ddca_create_display_ref(did,&ddca_dref);
+                        ddca_get_display_ref(did,&ddca_dref);
                         ddca_open_display2(ddca_dref,false,&display.ddca_dh_loc);
                         displayInfo_V.append(display);
                     }
